subsys/bluetooth/mesh/shell.c

```diff
@@ -3263,3 +3623,58 @@ static int cmd_mesh(const struct shell *shell, size_t argc, char **argv)
 
 SHELL_CMD_ARG_REGISTER(mesh, &mesh_cmds, "Bluetooth mesh shell commands",
 			cmd_mesh, 1, 1);
+
+
+long bt_mesh_shell_strtol(const char *str, int base, int *err)
+{
+	long val;
+	char *endptr = NULL;
+
+	errno = 0;
+	val = strtol(str, &endptr, base);
+	if (errno == ERANGE) {
+		*err = -ERANGE;
+		return 0;
+	} else if (errno || endptr == str || *endptr++) {
+		*err = -EINVAL;
+		return 0;
+	}
+
+	return val;
+}
+
+unsigned long bt_mesh_shell_strtoul(const char *str, int base, int *err)
+{
+	unsigned long val;
+	char *endptr = NULL;
+
+	if (*str == '-') {
+		*err = -EINVAL;
+		return 0;
+	}
+
+	errno = 0;
+	val = strtoul(str, &endptr, base);
+	if (errno == ERANGE) {
+		*err = -ERANGE;
+		return 0;
+	} else if (errno || endptr == str || *endptr++) {
+		*err = -EINVAL;
+		return 0;
+	}
+
+	return val;
+}
+
+bool bt_mesh_shell_strtobool(const char *str, int base, int *err)
+{
+	if (!strcmp(str, "on") || !strcmp(str, "enable") || !strcmp(str, "true")) {
+		return true;
+	}
+
+	if (!strcmp(str, "off") || !strcmp(str, "disable") || !strcmp(str, "false")) {
+		return false;
+	}
+
+	return (bool)bt_mesh_shell_strtol(str, base, err);
```

Comment:
The reason for using `bt_mesh_shell_strtol` is to allow e.g. `-1` to be interpreted as a `true` conversion rather than it being evaluated as a invalid input. It is a corner case from a user perspective, but I do not see any particular reason why it should be prohibited. Do you?

Reply:
I thought the conversion between bool and numeric required a cast. It apparently does not so i will remove it 👍

Reply:
> The reason for using `bt_mesh_shell_strtol` is to allow e.g. `-1` to be interpreted as a `true` conversion rather than it being evaluated as a invalid input. It is a corner case from a user perspective, but I do not see any particular reason why it should be prohibited. Do you?

I'm not sure that `-1` should be interpreted as `true`.

Reply:
Ok, I will change it